core/src/main/scala/com/softwaremill/kmq/redelivery/streams/StartMarkerStream.scala

```diff
@@ -0,0 +1,38 @@
+package com.softwaremill.kmq.redelivery.streams
+
+import akka.Done
+import akka.actor.ActorSystem
+import akka.kafka.ConsumerMessage.CommittableMessage
+import akka.kafka.scaladsl.Consumer.DrainingControl
+import akka.kafka.scaladsl.{Committer, Consumer, Producer}
+import akka.kafka._
+import akka.stream.scaladsl.Sink
+import com.softwaremill.kmq.{MarkerKey, MarkerValue, StartMarker}
+import org.apache.kafka.clients.producer.ProducerRecord
+
+class StartMarkerStream(msgConsumerSettings: ConsumerSettings[String, String],
+                        markerProducerSettings: ProducerSettings[MarkerKey, MarkerValue],
+                        queueTopic: String, markersTopic: String, maxPartitions: Int, messageTimeout: Long)
+                       (implicit system: ActorSystem) {
+
+  def run(): DrainingControl[Done] = {
+    val committerSettings = CommitterSettings(system)
+
+    Consumer.committablePartitionedSource(msgConsumerSettings, Subscriptions.topics(queueTopic))
+      .mapAsyncUnordered(maxPartitions) {
```

Comment:
TBH this stream will be so fast that we don't need a partitioned source, I would go with a committableSource and .mapAsync(ConfigurableParallelism) (with a default value =1)

Reply:
After discussion: we agreed to leave it as-is for now.